core/controlplane/config/templates/cloud-config-worker

```diff
@@ -693,6 +659,74 @@ write_files:
       rkt rm --uuid-file=/var/run/coreos/set-aws-environment.uuid || :
 {{end}}
 
+  {{if .Experimental.AwsNodeLabels.Enabled -}}
+  - path: /opt/bin/kube-node-label
+    permissions: 0700
+    owner: root:root
+    content: |
+      #!/bin/bash -e
+      set -ue
+
+      INSTANCE_ID="$(/usr/bin/curl -s http://169.254.169.254/latest/meta-data/instance-id)"
+      SECURITY_GROUPS="$(/usr/bin/curl -s http://169.254.169.254/latest/meta-data/security-groups | tr '\n' ',')"
+      {{if not .SpotFleet.Enabled -}}
+      AUTOSCALINGGROUP="$(/usr/bin/docker run --rm --net=host \
+        {{.AWSCliImage.RepoWithTag}} aws \
+        autoscaling describe-auto-scaling-instances \
+        --instance-ids ${INSTANCE_ID} --region {{.Region}} \
+        --query 'AutoScalingInstances[].AutoScalingGroupName' --output text)"
+      LAUNCHCONFIGURATION="$(/usr/bin/docker run --rm --net=host \
+        {{.AWSCliImage.RepoWithTag}} \
+        aws autoscaling describe-auto-scaling-groups \
+        --auto-scaling-group-name $AUTOSCALINGGROUP --region {{.Region}} \
+        --query 'AutoScalingGroups[].LaunchConfigurationName' --output text)"
+      {{end -}}
+
+      label() {
+        /usr/bin/docker run --rm -t --net=host \
+          -v /etc/kubernetes:/etc/kubernetes \
+          -v /etc/resolv.conf:/etc/resolv.conf \
+          -e INSTANCE_ID=${INSTANCE_ID} \
+          -e SECURITY_GROUPS=${SECURITY_GROUPS} \
+          {{if not .SpotFleet.Enabled -}}
+          -e AUTOSCALINGGROUP=${AUTOSCALINGGROUP} \
+          -e LAUNCHCONFIGURATION=${LAUNCHCONFIGURATION} \
+          {{end -}}
+          {{.HyperkubeImage.RepoWithTag}} /bin/bash \
+            -ec 'echo "placing labels and annotations with additional AWS parameters."; \
+             kctl="/kubectl --server={{.APIEndpointURL}}:443 --kubeconfig=/etc/kubernetes/kubeconfig/worker.yaml"; \
+             kctl_label="$kctl label --overwrite nodes/$(hostname)"; \
+             kctl_annotate="$kctl annotate --overwrite nodes/$(hostname)"; \
+             {{if not .SpotFleet.Enabled -}}
+             $kctl_label kube-aws.coreos.com/autoscalinggroup=${AUTOSCALINGGROUP}; \
+             $kctl_label kube-aws.coreos.com/launchconfiguration=${LAUNCHCONFIGURATION}; \
+             {{end -}}
+             $kctl_annotate kube-aws.coreos.com/securitygroups=${SECURITY_GROUPS}; \
+             echo "done."'
```

Comment:
I've verified the retry logic to work by injecting an inexistent command here and running `sudo /opt/bin/kube-node-label` on a worker node.